src/test/codewhisperer/service/keyStrokeHandler.test.ts

```diff
@@ -200,31 +190,26 @@ describe('keyStrokeHandler', function () {
             assert.ok(!invokeSpy.called)
         })
 
-        it('Should call invokeAutomatedTrigger with arg KeyStrokeCount when invocationContext.keyStrokeCount reaches threshold', async function () {
+        it('Should not start idle trigger timer when inputing special characters', async function () {
             const mockEditor = createMockTextEditor()
             const mockEvent: vscode.TextDocumentChangeEvent = createTextDocumentChangeEvent(
                 mockEditor.document,
                 new vscode.Range(new vscode.Position(0, 0), new vscode.Position(0, 1)),
-                'a'
+                '('
             )
-            KeyStrokeHandler.instance.keyStrokeCount = 15
             await KeyStrokeHandler.instance.processKeyStroke(mockEvent, mockEditor, mockClient, config)
-            invokeSpy('KeyStrokeCount', mockEditor, mockClient)
-            assert.ok(invokeSpy.called)
+            assert.ok(!startTimerSpy.called)
```

Comment:
This is just feedback for future implementations:

One way to avoid spies is to make the thing your testing more "visible". For example, the goal here is to test whether or not the timer is running right? So we could make that a field on `KeyStrokeHandler` and implement a bit more logic to make it work correctly. Usually the problem (benefit?) with doing this is that more code will be executed which means more things need to be written in a testable way. But this is generally my thought-process when writing tests.